### PR TITLE
Fix bug for statistical metrics is incorrect

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>
/kind improvement

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/indexcoord/meta_table.go [hunk 1]
```diff
@@ -905,6 +936,10 @@ func (mt *metaTable) RemoveIndex(collID, indexID UniqueID) error {
 	}
 
 	delete(mt.collectionIndexes[collID], indexID)
+	if len(mt.collectionIndexes[collID]) == 0 {
+		delete(mt.collectionIndexes, collID)
+		metrics.IndexCoordIndexTaskNum.DeleteLabelValues(strconv.FormatInt(collID, 10))
```
Comment (LoveEachDay) [inline internal/indexcoord/meta_table.go:941]: Here, should we reset the guage to zero?

Comment (xiaocai2333) [reply internal/indexcoord/meta_table.go:941]: there is no longer a chance to clean up these labels, and the metrics data collected by prometheus will get bigger and bigger.

**Pull request merged**